File: API/Models/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;

namespace API.Models.OrderAggregate
{
    public class Order
    {
        public Order()
        {
        }

        public Order(List<OrderItem> orderItems, string byerEmail, Address shippingAddress, Delivery delivery, int subtotal,
        string paymentIntentId)
        {
            ByerEmail = byerEmail;
            ShippingAddress = shippingAddress;
            Delivery = delivery;
            OrderItems = orderItems;
            Subtotal = subtotal;
            PaymentIntentId = paymentIntentId;
        }

        public int Id { get; set; }
        public string ByerEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public Address ShippingAddress { get; set; }
        public Delivery Delivery { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public int Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.NaCekanju;
        public string PaymentIntentId { get; set; }

        public int GetTotal()
        {
            return Subtotal + Delivery.Price;
        }
    }
}

File: client/src/app/_models/order.ts
import { Address } from "cluster"

export interface IOrderToCreate {
    cartId: string
    deliveryId: number
    shippingAddress: Address
}



export interface Delivery {
    id: number;
    shortName: string;
    deliveryTime: string;
    description: string;
    price: number;
}

export interface ItemOrdered {
    productItemId: number;
    productName: string;
    pictureUrl: string;
}

export interface OrderItem {
    id: number;
    itemOrdered: ItemOrdered;
    price: number;
    quantity: number;
}

export interface IOrder {
    id: number;
    byerEmail: string;
    orderDate: string;
    shippingAddress: Address;
    delivery: Delivery;
    orderItems: OrderItem[];
    subtotal: number;
    status: number;
    paymentIntentId?: any;
}
